src/language-handlebars/printer-glimmer.js

```diff
@@ -764,11 +770,67 @@ function printBlockParams(node) {
   return ["as |", node.blockParams.join(" "), "|"];
 }
 
+function getCssParser(node, options) {
+  if (node.tag !== "style") {
+    return;
+  }
+
+  return inferParser(options, { language: "css" });
+}
+
+function getNodeContent(node, options) {
+  if (node.children.length === 0) {
+    return "";
+  }
+  return options.originalText.slice(
+    node.children.at(0).loc.start.offset,
+    node.children.at(-1).loc.end.offset,
+  );
+}
+
+function embed(path, options) {
+  const { node } = path;
+
+  if (node.type === "ElementNode") {
```

Comment:
I think we should check `TextNode` instead, `<style>` should not printed here.

Reply:
I'm not sure why, can you explain? The CSS content from `style` tags comes as `TextNode` in the `embed` function.

Reply:
Because we may have tag print logic in the "generic" print. We should only care about CSS code.

Reply:
Fixed in https://github.com/prettier/prettier/pull/15087/commits/6804b38b0c4d271758b95d97284e9615de37fe2d